spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/properties/KafkaBinderConfigurationProperties.java

```diff
@@ -559,6 +565,17 @@ public void setProducerProperties(Map<String, String> producerProperties) {
 				ProducerConfig.BOOTSTRAP_SERVERS_CONFIG);
 	}
 
+	private void filterStreamManagedConfiguration(Map<String, Object> configuration) {
+		Stream.of(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, ConsumerConfig.GROUP_ID_CONFIG)
+				.forEach(streamManagedConfiguration -> {
+					if (configuration.containsKey(streamManagedConfiguration)) {
+						logger.warn("Ignoring provided value(s) for '" + streamManagedConfiguration + "'. " +
+								"This configuration (and related functionality) is managed by Spring Cloud Stream.");
+						configuration.remove(streamManagedConfiguration);
```

Comment:
Given that some might consider this a breaking change (although we generally don't consider a fix that breaks code that relyies on incorrect behavior [bug] as "breaking"), I think we need more specific warning logs.

I also think we should only warn if they explicitly set auto-commit to 'true', setting it to false is ok.

`ConsumerConfig. ENABLE_AUTO_COMMIT_CONFIG + "=true is not supported by the Kafka binder`
`Use spring.cloud.stream.binding.<name>.group to specify the group instead of " + ConsumerConfig.GROUP_ID_CONFIG`

Reply:
@garyrussell I was trying to keep the warning generic just in case more configuration is added to the set of ignored/overridden configuration but I've now updated the PR with more explicit warnings based on your suggestions.